src/mono/mono/metadata/metadata.c

```diff
@@ -2575,13 +2576,13 @@ mono_metadata_signature_dup_delegate_invoke_to_target (MonoMethodSignature *sig)
  * @return the new \c MonoMethodSignature structure.
  */
 MonoMethodSignature*
-mono_metadata_signature_dup_new_params (MonoMemPool *mp, MonoMemoryManager *mem_manager, MonoMethodSignature *sig, uint32_t num_params, MonoType **new_params)
+mono_metadata_signature_dup_new_params (MonoMemPool *mp, MonoMemoryManager *mem_manager, MonoImage *image, MonoMethodSignature *sig, uint32_t num_params, MonoType **new_params)
```

Comment:
I think we should try to move away from passing `MonoImage*` to allocation functions.  you can pass the mempool of the image if you need to.  or better to get the mem manager for the method definition, I think.

Reply:
Reversed the change. Just for my education, why is passing `MonoImage*` a code smell?